File: VB/ComplexSummary/XtraReport1.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.XtraReports.UI
' ...

Namespace ComplexSummary
	Partial Public Class XtraReport1
		Inherits XtraReport
		Public Sub New()
			InitializeComponent()
		End Sub

		Public avgPrice As Double
		Public sumUnits As Double
		Public total As Double

		Private Sub xrLabel5_SummaryCalculated(ByVal sender As Object, ByVal e As TextFormatEventArgs) Handles xrLabel5.SummaryCalculated
			avgPrice += Convert.ToDouble(e.Value)
		End Sub

		Private Sub xrLabel6_SummaryCalculated(ByVal sender As Object, ByVal e As TextFormatEventArgs) Handles xrLabel6.SummaryCalculated
			sumUnits += Convert.ToDouble(e.Value)
		End Sub

		Private Sub xrLabel7_PrintOnPage(ByVal sender As Object, ByVal e As PrintOnPageEventArgs) Handles xrLabel7.PrintOnPage
			total = avgPrice + sumUnits
			CType(sender, XRLabel).Text = total.ToString("#,#")
		End Sub

	End Class
End Namespace

File: CS/ComplexSummary/XtraReport1.cs
using System;
using DevExpress.XtraReports.UI;
// ...

namespace ComplexSummary {
    public partial class XtraReport1 : XtraReport {
        public XtraReport1() {
            InitializeComponent();
        }

        public double avgPrice;
        public double sumUnits;
        public double total;

        private void xrLabel5_SummaryCalculated(object sender, TextFormatEventArgs e) {
            avgPrice += Convert.ToDouble(e.Value);
        }

        private void xrLabel6_SummaryCalculated(object sender, TextFormatEventArgs e) {
            sumUnits += Convert.ToDouble(e.Value);
        }

        private void xrLabel7_PrintOnPage(object sender, PrintOnPageEventArgs e) {
            total = avgPrice + sumUnits;
            ((XRLabel)sender).Text = total.ToString("#,#");
        }

    }
}
